src/cargo/util/config/mod.rs

```diff
@@ -545,9 +561,27 @@ impl Config {
                 return true;
             }
         }
+        self.check_environment_key_case_mismatch(key);
+
         false
     }
 
+    fn check_environment_key_case_mismatch(&self, key: &ConfigKey) {
+        if cfg!(windows) {
+            return;
+        }
+        match self.upper_case_env.get(key.as_env_key()) {
+            Some(env_key) => {
+                let _ = self.shell().warn(format!(
+                    "Variables in environment require uppercase,
```

Comment:
This needs a backslash to continue on the next line.

```suggestion
                    "Variables in environment require uppercase, \
```